eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/Relayer.scala

```diff
@@ -149,10 +149,10 @@ class Relayer(nodeParams: NodeParams, router: ActorRef, register: ActorRef, comm
 
     case Status.Failure(addFailed: AddHtlcFailed) =>
       addFailed.origin match {
-        case Origin.Local(id, None) => log.error(s"received unexpected add failed with no sender (paymentId=$id)")
+        case Origin.Local(id, None) => postRestartCleaner forward addFailed
```

Comment:
Exactly, I've always felt uneasy with those unhandled cases, I knew I was missing something but I didn't know what...Now I know, and I'll sleep better xD

Reply:
They're back and still not handled now, but this is because they really should never appear now that `AddHtlcFailed` is used properly...still feels dangerous though, I don't know how I could improve that.